instrumentation/jetty-httpclient/jetty-httpclient-9.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_0/JettyHttpClient9InstrumentationModule.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_0;
+
+import static io.opentelemetry.javaagent.extension.matcher.ClassLoaderMatcher.hasClassesNamed;
+import static io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge.currentContext;
+import static io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_0.JettyHttpClient9Tracer.tracer;
+import static java.util.Collections.singletonList;
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import java.util.List;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.eclipse.jetty.client.api.Request;
+
+@AutoService(InstrumentationModule.class)
+public class JettyHttpClient9InstrumentationModule extends InstrumentationModule {
+
+  public JettyHttpClient9InstrumentationModule() {
+    super("jetty-httpclient", "jetty-httpclient-9");
+  }
+
+  @Override
+  public List<TypeInstrumentation> typeInstrumentations() {
+    return singletonList(new JettyHttpClient9Instrumentation());
+  }
+
+  public static class JettyHttpClient9Instrumentation implements TypeInstrumentation {
+
+    @Override
+    public ElementMatcher<ClassLoader> classLoaderOptimization() {
+      return hasClassesNamed("org.eclipse.jetty.client.api.Request$BeginListener");
+    }
+
+    @Override
+    public ElementMatcher<TypeDescription> typeMatcher() {
+      return hasSuperType(named("org.eclipse.jetty.client.api.Request").and(isInterface()));
+    }
+
+    @Override
+    public void transform(TypeTransformer transformer) {
+      transformer.applyAdviceToMethod(
+          isMethod().and(named("send")),
```

Comment:
It looks like this just delegates to `HttpClient.send` I think that's a better instrumentation point. To get the context to propagate, we need to wrap the `List<Response.ResponseListener>` callbacks with context. For example, something like

```java
List<Response.ResponseListener> wrapped = listeners.stream()
  .map(listener -> {
    if (listener instanceof Response.CompleteListener) {
      return (Response.CompleteListener) result -> {
        try (Scope ignored = context.makeCurrent()) {
          listener.onComplete(result);
        }
      }
  .collect(toList());
super.send(request, wrapped);
```

Normally we want all callbacks to be wrapped with context, but jetty has a ton of types so that would be very tedious...

https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/client/api/Response.ResponseListener.html

`CompleteListener` seems to be the most important one.

Reply:
Response listener wrapping is added now